scheduler/scheduler.py

```diff
@@ -2666,24 +2669,34 @@ def _update_lease_callback(self, job_id, worker_id, steps, duration,
                                                         max_steps,
                                                         max_duration))
 
-        # Round the remaining steps to the nearest multiple of scale_factor.
-        scale_factor = self._jobs[job_id].scale_factor
-        remaining_steps = self._get_remaining_steps(job_id)
-        remaining_steps = int(math.ceil(remaining_steps / scale_factor))
+            # Round the remaining steps to the nearest multiple of scale_factor.
+            scale_factor = self._jobs[job_id].scale_factor
+            remaining_steps = self._get_remaining_steps(job_id)
+            remaining_steps = int(math.ceil(remaining_steps / scale_factor))
+            current_time = self.get_current_timestamp()
+            current_round_end_time = \
+                self._current_round_start_time + self._time_per_iteration
+            remaining_time_in_current_round = \
+                current_round_end_time - current_time
+            remaining_time_in_current_round = \
+                max(0, remaining_time_in_current_round)
 
         if steps == 0 or duration == 0:
-            return (remaining_steps, self._time_per_iteration)
+            return (remaining_steps, remaining_time_in_current_round)
 
         # Extend the lease if the job has been placed on the same workers
         # for the upcoming round.
         with self._scheduler_lock:
             # TODO: Remove scan of self._jobs_with_extended_lease.
             for job_id_combination in self._jobs_with_extended_lease:
                 if job_id.overlaps_with(job_id_combination):
-                    return (max_steps, max_duration + self._time_per_iteration)
+                    updated_lease_duration = max_duration
```

Comment:
What is `max_duration`? Maximum duration across tasks?

Reply:
`max_duration` is the lease duration